app/src/main/java/org/kiwix/kiwixmobile/intro/IntroActivity.kt

```diff
@@ -0,0 +1,142 @@
+import org.kiwix.kiwixmobile.intro.IntroContract.View
+import org.kiwix.kiwixmobile.kiwixActivityComponent
+import java.util.Timer
+import java.util.TimerTask
+
+const val DURATION_WAITER = 800
+const val PERIOD_TIME = 2000
+const val DELAY_TIME = 0
+
+class IntroActivity : BaseActivity(), View {
+  private val handler = Handler()
+  private val timer = Timer()
+  @BindView(id.view_pager)
+  @Nullable
+  @JvmField
+  var viewPager: ViewPager? = null
+  @BindView(id.tab_indicator)
+  @Nullable
+  @JvmField
+  var tabIndicator: InkPageIndicator? = null
+  var presenter: Presenter? = null
+  private var airPlane: ImageView? = null
+  private var currentPage = 0
+  private val pageChangeListener: OnPageChangeListener = object : OnPageChangeListener {
+    override fun onPageScrolled(
+      position: Int,
+      positionOffset: Float,
+      positionOffsetPixels: Int
+    ) {
+      Log.d("CustomViewPager", "pageChangeListener")
+    }
+
+    override fun onPageSelected(position: Int) {
+      if (position == 1) {
+        airPlane!!.visibility = android.view.View.VISIBLE
+        airPlane!!.animate()
+          .translationX(airPlane!!.width.toFloat()).duration = DURATION_WAITER.toLong()
+      } else {
+        airPlane!!.visibility = android.view.View.INVISIBLE
+        airPlane!!.animate()
+          .translationX(-airPlane!!.width.toFloat())
+      }
+      currentPage = position
+    }
+
+    override fun onPageScrollStateChanged(state: Int) {
+      if (state == ViewPager.SCROLL_STATE_DRAGGING) {
+        dismissAutoRotate()
+      }
+    }
+  }
+  private lateinit var views: Array<android.view.View>
+  override fun injection(coreComponent: CoreComponent) {
+    this.kiwixActivityComponent.inject(this)
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(layout.activity_intro)
+    val layoutInflater = layoutInflater
+    views = arrayOf(
+      layoutInflater.inflate(layout.item_intro_1, viewPager, false),
+      layoutInflater.inflate(layout.item_intro_2, viewPager, false)
+    )
+    val introPagerAdapter = IntroPagerAdapter(views)
+    viewPager!!.adapter = introPagerAdapter
+    tabIndicator!!.setViewPager(viewPager)
+    airPlane = views[1].findViewById(id.airplane)
+    viewPager!!.addOnPageChangeListener(pageChangeListener)
+    timer.schedule(object : TimerTask() {
+      override fun run() {
+        handler.post {
+          if (currentPage == views.size) {
+            currentPage = 0
+          }
+          viewPager!!.setCurrentItem(currentPage++, true)
+        }
+      }
+    }, DELAY_TIME.toLong(), PERIOD_TIME.toLong())
+    for (view in views) {
+      view.findViewById<android.view.View>(id.root)
+        .setOnClickListener { v: android.view.View? -> dismissAutoRotate() }
+    }
+  }
+
+  override fun onDestroy() {
+    super.onDestroy()
+    handler.removeCallbacksAndMessages(null)
+    timer.cancel()
+    for (view in views) {
```

Comment:
this was silly in the original incarnation, you don't have to remove onClickListeners for the activity that is getting destroyed